File: Medium/971. Flip Binary Tree To Match Preorder Traversal/RecursiveSolution.java
class Solution {
    int x = 0;

    public List<Integer> flipMatchVoyage(TreeNode root, int[] voyage) {
        List<Integer> res = new ArrayList();
        return visit(root, voyage, res) ? res : new ArrayList(Arrays.asList(-1));
    }

    private boolean visit(TreeNode node, int[] v, List<Integer> res) {
        if (node == null)
            return true;

        if (x >= v.length || node.val != v[x])
            return false;

        x++;
        if (node.left == null || node.left.val == v[x])
            return visit(node.left, v, res) && visit(node.right, v, res);

        res.add(node.val);
        return visit(node.right, v, res) && visit(node.left, v, res);
    }
}

File: Medium/230. Kth Smallest Element in a BST/recursive-solution.js
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthSmallest = function (root, k) {
  const res = [];
  traverse(root);
  return res[k - 1];

  function traverse(node) {
    if (!node) return;
    traverse(node.left);
    res.push(node.val);
    traverse(node.right);
  }
};
